### PR TITLE
Remove the always-enabled login feature flags

### PR DESCRIPTION
**Pull request by jsnajdr**
Remove two feature flags that are always true since desktop build was removed: `login/wp-login` and `login/native-login-links`.

### CODE REVIEW

**Review by sarayourfriend — APPROVED** [review]
LGTM. Only question is whether we should remove the `isNative` boolean altogether given that it's always true now (if that is indeed the case that it is always true?)

**Comment by jsnajdr** [comment]
> we should remove the isNative boolean altogether given that it's always true now

Good idea! Doing that in a79ffe7

> (if that is indeed the case that it is always true?)

It's not, there were some `login()` calls without the `isNative` arg. Many of them are just omissions, e.g., in the "accept an invite" pages. These would navigate to `wp-login.php`, which, however, immediately redirects to the Calypso `/log-in`. So, assuming `isNative` there is OK.

Then there were account recovery links to `wp-login.php?action=lostpassword` constructed with `addQueryArgs( { action: 'lostpassword' }, login() )`. These must continue to point to `wp-login.php`. `/log-in?action=lostpassword` would not work. I replaced them with something like:
```
localizeUrl( 'https://wordpress.com/wp-login.php?action=lostpassword', locale )
```

To summarize, removing `isNative` is a bit of non-trivial work, but it was worth it.

File: client/blocks/signup-form/index.jsx [hunk 1]
```diff
@@ -424,7 +419,7 @@ class SignupForm extends Component {
 		return login( {
 			isJetpack: this.isJetpack(),
 			from: this.props.from,
-			isNative: config.isEnabled( 'login/native-login-links' ),
+			isNative: true,
```
Comment (tyxla) [inline client/blocks/signup-form/index.jsx]: We can get rid of `isNative` here, no?

File: client/blocks/signup-form/index.jsx [hunk 2]
```diff
@@ -928,9 +921,7 @@ class SignupForm extends Component {
 				'socialServiceResponse',
 			] );
 
-			const logInUrl = config.isEnabled( 'login/native-login-links' )
-				? this.getLoginLink()
-				: localizeUrl( config( 'login_url' ), this.props.locale );
+			const logInUrl = this.getLoginLink();
```
Comment (tyxla) [inline client/blocks/signup-form/index.jsx]: Do we really need a variable for this anymore? Perhaps we can just reference `this.getLoginLink()` directly in this file?

File: client/jetpack-connect/signup.js [hunk 3]
```diff
@@ -380,18 +380,15 @@ export class JetpackSignup extends Component {
 				pageTitle = translate( 'Login to WordPress.com' );
 				footerLinks.push(
 					<LoggedOutFormLinkItem key="signup" onClick={ this.showWooDnaSignupView }>
-						{ this.props.translate( 'Create a new account' ) }
+						{ translate( 'Create a new account' ) }
```
Comment (tyxla) [inline client/jetpack-connect/signup.js:382]: 👍

File: client/jetpack-connect/signup.js [hunk 4]
```diff
@@ -380,18 +380,15 @@ export class JetpackSignup extends Component {
...
 					</LoggedOutFormLinkItem>
 				);
 				footerLinks.push(
 					<LoggedOutFormLinkItem
 						key="lostpassword"
-						href={ addQueryArgs(
-							{ action: 'lostpassword' },
-							login( { locale: this.props.locale } )
-						) }
+						href={ localizeUrl( 'https://wordpress.com/wp-login.php?action=lostpassword', locale ) }
```
Comment (tyxla) [inline client/jetpack-connect/signup.js]: Could you please elaborate on this change? I'm fine with the use of `localizeUrl()` but don't see a good reason to hardcode the login URL. 

For the protocol, I've verified this works the same way as before.

To check that:

* Go to `/jetpack/connect/authorize/es?client_id=123123&site=example.com&nonce=nonce&secret=1&_wp_nonce=something&blogname=something&home_url=something&redirect_uri=something&scope=something&secret=something&site=something&site_url=something&state=something&woodna_service_name=login&debug`
* In the console, type `dispatch( { type: 'LOGIN_AUTH_ACCOUNT_TYPE_REQUEST_SUCCESS', data: { type: 'regular' } } )`
* Observe the URL of the forgotten password link (the second of three footer links). It should have the localized subdomain and properly link to the WP core forgotten password page on that language.

File: config/production.json [hunk 5]
```diff
@@ -77,8 +77,6 @@
 		"layout/query-selected-editor": true,
 		"legal-updates-banner": false,
 		"login/magic-login": true,
-		"login/native-login-links": true,
```
Comment (tyxla) [inline config/production.json:80]: I see these aren't enabled for Jetpack Cloud. Any concerns with that?

File: client/login/magic-login/emailed-login-link-expired.jsx [hunk 6]
```diff
@@ -21,14 +20,8 @@ import {
...
-const nativeLoginUrl = login( { isNative: true, twoFactorAuthType: 'link' } );
-
-const lostPasswordURL = addQueryArgs(
-	{
-		action: 'lostpassword',
-	},
-	login()
-);
+const nativeLoginUrl = login( { twoFactorAuthType: 'link' } );
+const lostPasswordUrl = 'https://wordpress.com/wp-login.php?action=lostpassword';
```
Comment (tyxla) [inline client/login/magic-login/emailed-login-link-expired.jsx]: I'm really unhappy with us hardcoding this URL here and there. I think we should either create a new config var for it or just use `config( 'login_url' )` with the query arg.

File: client/login/wp-login/login-links.jsx [hunk 7]
```diff
@@ -74,13 +75,7 @@ export class LoginLinks extends React.Component {
...
 		this.props.recordTracksEvent( 'calypso_login_lost_phone_link_click' );
 
-		page(
-			login( {
-				isNative: true,
-				twoFactorAuthType: 'backup',
-				isGutenboarding: this.props.isGutenboarding,
-			} )
-		);
+		page( login( { twoFactorAuthType: 'backup', isGutenboarding: this.props.isGutenboarding } ) );
```
Comment (tyxla) [inline client/login/wp-login/login-links.jsx:77]: There's another instance of `isNative` in this file; should we remove it as well?

**Review by tyxla — COMMENTED** [review]
Great cleanup, Jarda, love it! 👍 

I've left some questions and comments, let me know what you think.

File: client/blocks/signup-form/index.jsx (hunk 1 above)
Comment (jsnajdr) [reply client/blocks/signup-form/index.jsx]: Fixed, including inlining the variables into JSX in this file.

File: client/jetpack-connect/signup.js (hunk 4 above)
Comment (jsnajdr) [reply client/jetpack-connect/signup.js]: Well this is not really a login URL, but URL of the "lost password" page. Whose URL happens to be `wp-login.php?action=lostpassword`. This page was never implemented in Calypso (I think there was a project for it that never shipped). Going to `/log-in?action=lostpassword` won't redirect to the "lost password" page.

So, this is a completely separate page that has little to do with the `login()` helper. As it's used at several places, I'll extract it into a helper on its own.

File: config/production.json (hunk 5 above)
Comment (jsnajdr) [reply config/production.json:80]: That's a good point, I forgot about Jetpack Cloud 🙂  But I now verified that the `login` helper is not used in any Jetpack Cloud code.

For OAuth-enabled environments, we'll need to carefully inspect where we are redirecting in "redirect to login if logged out" scenarios. Instead of `/log-in`, it's more proper to redirect to the `wordpress.com/oauth2/authorize` page. But that can be done in a different PR.

File: client/jetpack-connect/signup.js (hunk 4 above)
Comment (tyxla) [reply client/jetpack-connect/signup.js]: A separate helper sounds good to me in this instance 👍

**Comment by jsnajdr** [comment]
@tyxla I addressed your review feedback: removed some remaining usages of `isNative`, and created a `lostPassword` path help. Let me know what you think.

File: client/blocks/signup-form/index.jsx [hunk 8]
```diff
@@ -486,12 +479,10 @@ class SignupForm extends Component {
 			return;
 		}
 
-		let link = this.getLoginLink();
-
 		return map( messages, ( message, error_code ) => {
 			if ( error_code === 'taken' ) {
 				const fieldValue = formState.getFieldValue( this.state.form, fieldName );
-				link += '&email_address=' + encodeURIComponent( fieldValue );
+				const link = addQueryArgs( { email_address: fieldValue }, this.getLoginLink() );
```
Comment (tyxla) [inline client/blocks/signup-form/index.jsx:485]: 👍

File: client/lib/paths/lost-password.ts [hunk 9]
```diff
@@ -0,0 +1,8 @@
+/**
+ * Internal dependencies
+ */
+import { localizeUrl } from 'calypso/lib/i18n-utils';
+
+export function lostPassword( locale?: string ): string {
```
Comment (tyxla) [inline client/lib/paths/lost-password.ts]: `login()` accepts an object of args, while `lostPassword()` a single arg. Are we fine with that little discrepancy? I'm personally okay with it, just wanted to confirm your intentions.

Comment (tyxla) [inline client/lib/paths/lost-password.ts]: While in practice `locale` is a `string`, it's more precise to use the `Locale`  type, as declared [here](https://github.com/Automattic/wp-calypso/blob/14466513488e705acbce8a7b675024bda07d708b/packages/i18n-utils/src/locales.ts#L7).

File: client/login/magic-login/emailed-login-link-expired.jsx [hunk 10]
```diff
@@ -21,14 +20,8 @@ import {
...
-const nativeLoginUrl = login( { isNative: true, twoFactorAuthType: 'link' } );
-
-const lostPasswordURL = addQueryArgs(
-	{
-		action: 'lostpassword',
-	},
-	login()
-);
+const nativeLoginUrl = login( { twoFactorAuthType: 'link' } );
+const lostPasswordUrl = lostPassword();
```
Comment (tyxla) [inline client/login/magic-login/emailed-login-link-expired.jsx]: Seems like we don't need a constant for that one - it's used just once and is short enough to be used directly for the `secondaryActionURL` prop below.

File: client/state/login/utils.js [hunk 11]
```diff
@@ -66,22 +67,19 @@ export function getErrorFromHTTPError( httpError ) {
...
 		} else if ( code === 'compromisable_account' ) {
-			const url = localizeUrl( 'https://wordpress.com/wp-login.php?action=lostpassword' );
 			return {
 				code,
 				message: (
 					<p>
 						{ translate(
 							'Your account has been blocked as a security precaution. To continue, you must {{a}}reset your password{{/a}}.',
-							{ components: { a: <a href={ url } rel="external" /> } }
+							{ components: { a: <a href={ lostPassword() } rel="external" /> } }
```
Comment (tyxla) [inline client/state/login/utils.js:76]: Although this was like that before, current locale (if any) would be ignored that way as we're not taking it into consideration and not passing it further down to `localizeUrl()`. Should we fix that?

**Review by tyxla — APPROVED** [review]
This PR grew to be pretty big 😉 

It tests well and looks good to me, I've just left some non-blocking minor suggestions and questions. Feel free to ship either way.

Great cleanup!

File: client/lib/paths/lost-password.ts (hunk 9 above)
Comment (jsnajdr) [reply client/lib/paths/lost-password.ts]: Good idea, pushed a fix where the function API is `lostPassword( { locale } )`.

Comment (jsnajdr) [reply client/lib/paths/lost-password.ts]: `Locale` type is a good idea for languages with nominal types like Java, where `Locale` really means something different from `string`. But TypeScript types are structural and the `Locale` alias is much less useful. I think we should remove it and just use `string` everywhere.

File: client/login/magic-login/emailed-login-link-expired.jsx (hunk 10 above)
Comment (jsnajdr) [reply client/login/magic-login/emailed-login-link-expired.jsx]: Inlined that. And improved the `EmailedLoginLinkExpired` component even more: the `page( nativeLoginUrl )` call is not really needed.

File: client/state/login/utils.js (hunk 11 above)
Comment (jsnajdr) [reply client/state/login/utils.js:76]: `localizeUrl` reads the locale from `i18n.getLocaleSlug()` if the `locale` param is not provided. I think we should remove that hidden dependency, but in another PR.

File: client/lib/paths/lost-password.ts (hunk 9 above)
Comment (tyxla) [reply client/lib/paths/lost-password.ts]: > I think we should remove it and just use string everywhere.

I actually wouldn't mind that too much. I guess the idea behind keeping it a separate type would be if we want to get a bit stricter and allow only a limited set of actual locales there.

File: client/state/login/utils.js (hunk 11 above)
Comment (tyxla) [reply client/state/login/utils.js:76]: Sounds good to me 👍
